core/src/main/java/org/infinispan/container/offheap/OffHeapDataContainer.java

```diff
@@ -587,7 +587,7 @@ public void executeTask(KeyValueFilter<? super WrappedBytes, ? super WrappedByte
 
    @Override
    public Iterator<InternalCacheEntry<WrappedBytes, WrappedBytes>> iterator() {
-      long time = timeService.time();
+      long time = timeService.wallClockTime();
```

Comment:
This was the actual bug btw. So depending on the current system time it would show itself or not. Hence why I was never able to see the issue.